src/engine/effects/engineeffectchain.cpp

```diff
@@ -23,6 +24,8 @@ EngineEffectChain::EngineEffectChain(const QString& group,
         }
         m_chainStatusForChannelMatrix.insert(inputChannel.handle(), outputChannelMap);
     }
+
+    m_pEffectsDelay = std::make_unique<EngineEffectsDelay>();
```

Comment:
Maybe I am a little bit confused. What I think is to construct an object by which I use the class methods. Can you please describe your proposal or what solution you propose?

Reply:
Right. but what you are doing right now is that you are allocating an instance of `EngineEffectsDelay` on the heap and then storing the point in the `EngineEffectChain`. But you don't have to allocate on the heap. You can just construct the object as part of the owning object. 
Since `EngineEffectChain` lives in the audio engine thread, using heap-allocation is a nogo.
Now:
```
 ┌─────────────────┐       ┌─────────────────┐
 │EngineEffectChain│   ┌──►│EngineEffectDelay│
 │                 │   │   │                 │
 │m_pEffectDelay ──┼───┘   │...              │
 │                 │       └─────────────────┘
 │...              │
 └─────────────────┘
``` 
What I propose:
```

 ┌─────────────────────┐
 │EngineEffectChain    │
 │ ┌─────────────────┐ │
 │ │EngineEffectDelay│ │
 │ │                 │ │
 │ │...              │ │
 │ └─────────────────┘ │
 │...                  │
 └─────────────────────┘

```
I don't even think that the member would have to be constructed explicitly as it can be default constructed. So I'm basically just proposing to remove the line in question and change the class definition from `std::unique_ptr<EngineEffectsDelay> m_pEffectsDelay;` to `EngineEffectsDelay m_effectsDelay;`.

Reply:
Oh, clear. Thank you very much for this perfect explanation. It makes much more sense to me now.